staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go

```diff
@@ -268,3 +282,43 @@ func (f *FieldManager) stripFields(managed fieldpath.ManagedFields, manager stri
 
 	return managed
 }
+
+// NameDefaulter makes sure the request option and meta field match if both are set. If neither are set it defaults to the userAgent or errors
+func NameDefaulter(option, field, userAgent string) (string, error) {
+	if len(option) != 0 {
+		if len(field) != 0 && option != field {
+			return "", fmt.Errorf("fieldManager option must match metadata fieldManager or be unset")
+		}
+
+		return option, nil
+	}
+
+	if len(field) != 0 {
+		return field, nil
+	}
+
+	if len(userAgent) != 0 {
```

Comment:
Since this method is only used in two different places, and this condition only applies to one of them, I would have kept the comparison at the calling location (rather than have this method). Also, this method is also mixing "validation" at the same time. Do we know how this is done for namespaces?

Reply:
There seems to be a [ValidNamespace](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/registry/rest/meta.go#L37)

Reply:
Oh, cool.
Thanks for the ValidNamespace pointer.
I'll look into it and move things.